frontend/src/presentation/pages/QuestionsPage.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useState, useEffect, useCallback } from "react";
+import { Row, Col, Breadcrumb, Button, Spin, Alert } from "antd";
+import { QuestionList } from "../components/QuestionList";
+import { QuestionDetail } from "../components/QuestionDetail";
+import { LandingComponent } from "../components/LandingComponent";
+import { Question } from "../../domain/entities/Question";
+import { questionUseCases } from "../../domain/usecases/QuestionUseCases";
+import styles from "./QuestionsPage.module.css";
+import { ROUTES, ERRORS } from "presentation/utils/constants";
+import { handleError } from "presentation/utils/errorHandler";
+import { AddQuestionButton } from "presentation/components/buttons/AddQuestionButton";
+
+const LAST_SELECTED_QUESTION_KEY = "lastSelectedQuestionId";
+
+const QuestionsPage: React.FC = () => {
+  const [questions, setQuestions] = useState<Question[]>([]);
+  const [selectedQuestionId, setSelectedQuestionId] = useState<string | null>(
+    null
+  );
+  const [selectedQuestion, setSelectedQuestion] = useState<Question | null>(
+    null
+  );
+  const [isLoading, setIsLoading] = useState(true);
+  const [isQuestionLoading, setIsQuestionLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchQuestions = async () => {
+      setIsLoading(true);
+      try {
+        const fetchedQuestions = await questionUseCases.getAllQuestions();
+        setQuestions(fetchedQuestions);
+      } catch (err) {
+        setError(handleError(err, ERRORS.FAILED_TO_LOAD_QUESTIONS));
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchQuestions();
+
+    const lastSelectedQuestionId = localStorage.getItem(
+      LAST_SELECTED_QUESTION_KEY
```

Comment:
Hmm, I'm not quite understanding why localstorage is required here. Is it because we're using the same route for both cases when a user has not selected a question and when a user has selected a question?

Can we consider something like query params instead (e.g. /workspace?id=123)? Or slugs (e.g. /workspace/123). I think this might require more discussion so we can keep this now if we're short of time

Reply:
This use of local storage is here to avoid use of slugs because I'm still not entirely sure what they mean by an SPA, but I want the app to stay on the same question even if the user refreshes the page. That's all, really.

Reply:
As long as you're using React, you're already building a SPA since it only uses 1 html file. You can achieve "to stay on the same question even if the user refreshes the page" using slugs or query params too i think.